conans/client/generators/virtualenv.py

```diff
@@ -15,11 +15,15 @@ def get_setenv_variables_commands(deps_env_info, command_set=None):
         if platform.system() == "Windows":
             ret.append(command_set + ' "' + name + '=' + value + ';%' + name + '%"')
         else:
-            ret.append(command_set + ' ' + name + '=' + value + ':$' + name)
+            # Standard UNIX "sh" does not allow "export VAR=xxx" on one line
+            # So for portability reasons we split into two commands
+            ret.append(name + '=' + value + ':$' + name)
```

Comment:
I'm not fond of it but it turns out the `system` command in python invokes the default OS shell which is not guaranteed to handle Bash extensions.
For example on Solaris it's a vanilla `sh` that does not recognize `export VAR=xxx` in one line.